recommender/engines/autoencoders/training/model_training_step/model_training_step.py

```diff
@@ -0,0 +1,338 @@
+        self.user_batch_size = self.resolve_constant(USER_BATCH_SIZE, 128)
+        self.service_batch_size = self.resolve_constant(SERVICE_BATCH_SIZE, 128)
+        self.user_embedding_dim = self.resolve_constant(USER_EMBEDDING_DIM, 32)
+        self.service_embedding_dim = self.resolve_constant(SERVICE_EMBEDDING_DIM, 64)
+        self.learning_rate = self.resolve_constant(LR, 0.01)
+        self.epochs = self.resolve_constant(EPOCHS, 500)
+
+        self.trained_user_autoencoder_model = None
+        self.trained_services_autoencoder_model = None
+
+    def __call__(self, data=None) -> Tuple[object, dict]:  # TODO Refactor
+        """Perform user and service training"""
+
+        raw_data = data[AUTOENCODERS]
+
+        # Users
+        user_train_ds = raw_data[USERS][TRAIN]
+
+        user_train_ds_dl = DataLoader(
+            user_train_ds, batch_size=self.user_batch_size, shuffle=True
+        )
+
+        user_features_dim = len(
+            User.objects[0].one_hot_tensor
+        )  # TODO change the location of those
+        service_features_dim = len(Service.objects[0].one_hot_tensor)
+
+        user_model, user_loss, user_timer = perform_training(
+            collection_name=USERS,
+            train_ds_dl=user_train_ds_dl,
+            embedding_dim=self.user_embedding_dim,
+            features_dim=user_features_dim,
+            writer=self.writer,
+            device=self.device,
+            learning_rate=self.learning_rate,
+            epochs=self.epochs,
+            verbose=self.verbose,
+        )
+
+        # Services
+        service_train_ds = raw_data[SERVICES][TRAIN]
+
+        service_train_ds_dl = DataLoader(
+            service_train_ds, batch_size=self.service_batch_size, shuffle=True
+        )
+
+        service_model, service_loss, service_timer = perform_training(
+            collection_name=SERVICES,
+            train_ds_dl=service_train_ds_dl,
+            embedding_dim=self.service_embedding_dim,
+            features_dim=service_features_dim,
+            writer=self.writer,
+            device=self.device,
+            learning_rate=self.learning_rate,
+            epochs=self.epochs,
+            verbose=self.verbose,
+        )
+
+        details = {}
+        user_autoencoder_details = {LOSS: user_loss, TRAINING_TIME: user_timer}
+
+        service_autoencoder_details = {LOSS: service_loss, TRAINING_TIME: service_timer}
+
+        details[USERS] = user_autoencoder_details
+        details[SERVICES] = service_autoencoder_details
+
+        data = {}
+        user_autoencoder = {DATASET: raw_data[USERS], MODEL: user_model}
+
+        service_autoencoder = {DATASET: raw_data[SERVICES], MODEL: service_model}
+
+        data[USERS] = user_autoencoder
+        data[SERVICES] = service_autoencoder
+
+        return data, details
+
+    def save(self):
+        """Save a model"""
+        save_module(self.trained_user_autoencoder_model, name=USERS_AUTOENCODER)
```

Comment:
Autoencoders aren't products of the AEPipeline. Embedders are. They should be saved into DB (both for services and user - for now there is only saving for users). Embedders should be instantiate in the following way: `user_embedder = Embedder(user_autoencoder)` and saved into DB in the following way: `user_embedder.save(USER)` or `service_embedder.save(SERVICE)`